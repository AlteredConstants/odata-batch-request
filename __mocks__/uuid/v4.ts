module.exports = function uuidMock(): string {
  return "36522ad7-fc75-4b56-8c71-56071383e77b"
}
