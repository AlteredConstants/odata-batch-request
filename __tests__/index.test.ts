import * as _uuid from "uuid/v4"
import {
  ODataBatchChangeset,
  ODataBatchOperation,
  ODataBatchRequest,
} from "../src/index"

jest.mock("uuid/v4")

const uuid: jest.Mock<string, []> = _uuid as any

test("GET only", () => {
  uuid.mockReturnValue("36522ad7-fc75-4b56-8c71-56071383e77b")

  const batch = new ODataBatchRequest("host/service", [
    new ODataBatchOperation("get", "Customers('ALFKI')"),
    new ODataBatchOperation("get", "Products"),
  ])

  expect(batch.toString()).toMatchSnapshot()
})

test("GET and POST", () => {
  uuid.mockReturnValue("36522ad7-fc75-4b56-8c71-56071383e77b")

  const batch = new ODataBatchRequest("host/service", [
    new ODataBatchOperation("get", "Customers('ALFKI')"),
    new ODataBatchOperation("post", "Customers", {
      headers: { "Content-Type": "application/atom+xml;type=entry" },
      body: "<AtomPub representation of a new Customer>",
    }),
  ])

  expect(batch.toString()).toMatchSnapshot()
})

test("Changeset", () => {
  uuid
    // Changeset boundary.
    .mockReturnValueOnce("77162fcd-b8da-41ac-a9f8-9357efbbd")
    // Batch boundary.
    .mockReturnValueOnce("36522ad7-fc75-4b56-8c71-56071383e77b")

  const changeset = new ODataBatchChangeset([
    new ODataBatchOperation("post", "Customers", {
      headers: { "Content-Type": "application/atom+xml;type=entry" },
      body: "<AtomPub representation of a new Customer>",
    }),
    new ODataBatchOperation("patch", "Customers('ALFKI')", {
      headers: { "Content-Type": "application/json" },
      body: "<JSON representation of Customer ALFKI>",
    }),
  ])

  const batch = new ODataBatchRequest("host/service", [
    new ODataBatchOperation("get", "Customers('ALFKI')"),
    changeset,
    new ODataBatchOperation("get", "Products"),
  ])

  expect(batch.toString()).toMatchSnapshot()
})

test("Construct correct batch URL", () => {
  uuid.mockReturnValue("36522ad7-fc75-4b56-8c71-56071383e77b")

  const batch = new ODataBatchRequest("host/service/", [
    new ODataBatchOperation("get", "Customers"),
  ])

  expect(batch.toString()).toMatchSnapshot()
})
