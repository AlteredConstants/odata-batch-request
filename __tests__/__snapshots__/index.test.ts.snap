// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Changeset 1`] = `
"POST host/service/$batch HTTP/1.1
OData-Version: 4.0
Content-Type: multipart/mixed; boundary=batch_36522ad7-fc75-4b56-8c71-56071383e77b
Accept: multipart/mixed

--batch_36522ad7-fc75-4b56-8c71-56071383e77b
Content-Type: application/http
Content-Transfer-Encoding: binary

GET Customers('ALFKI') HTTP/1.1


--batch_36522ad7-fc75-4b56-8c71-56071383e77b
Content-Type: multipart/mixed; boundary=changeset_77162fcd-b8da-41ac-a9f8-9357efbbd

--changeset_77162fcd-b8da-41ac-a9f8-9357efbbd
Content-ID: 1
Content-Type: application/http
Content-Transfer-Encoding: binary

POST Customers HTTP/1.1
Content-Type: application/atom+xml;type=entry

<AtomPub representation of a new Customer>
--changeset_77162fcd-b8da-41ac-a9f8-9357efbbd
Content-ID: 2
Content-Type: application/http
Content-Transfer-Encoding: binary

PATCH Customers('ALFKI') HTTP/1.1
Content-Type: application/json

<JSON representation of Customer ALFKI>
--changeset_77162fcd-b8da-41ac-a9f8-9357efbbd--
--batch_36522ad7-fc75-4b56-8c71-56071383e77b
Content-Type: application/http
Content-Transfer-Encoding: binary

GET Products HTTP/1.1


--batch_36522ad7-fc75-4b56-8c71-56071383e77b--"
`;

exports[`Construct correct batch URL 1`] = `
"POST host/service/$batch HTTP/1.1
OData-Version: 4.0
Content-Type: multipart/mixed; boundary=batch_36522ad7-fc75-4b56-8c71-56071383e77b
Accept: multipart/mixed

--batch_36522ad7-fc75-4b56-8c71-56071383e77b
Content-Type: application/http
Content-Transfer-Encoding: binary

GET Customers HTTP/1.1


--batch_36522ad7-fc75-4b56-8c71-56071383e77b--"
`;

exports[`GET and POST 1`] = `
"POST host/service/$batch HTTP/1.1
OData-Version: 4.0
Content-Type: multipart/mixed; boundary=batch_36522ad7-fc75-4b56-8c71-56071383e77b
Accept: multipart/mixed

--batch_36522ad7-fc75-4b56-8c71-56071383e77b
Content-Type: application/http
Content-Transfer-Encoding: binary

GET Customers('ALFKI') HTTP/1.1


--batch_36522ad7-fc75-4b56-8c71-56071383e77b
Content-Type: application/http
Content-Transfer-Encoding: binary

POST Customers HTTP/1.1
Content-Type: application/atom+xml;type=entry

<AtomPub representation of a new Customer>
--batch_36522ad7-fc75-4b56-8c71-56071383e77b--"
`;

exports[`GET only 1`] = `
"POST host/service/$batch HTTP/1.1
OData-Version: 4.0
Content-Type: multipart/mixed; boundary=batch_36522ad7-fc75-4b56-8c71-56071383e77b
Accept: multipart/mixed

--batch_36522ad7-fc75-4b56-8c71-56071383e77b
Content-Type: application/http
Content-Transfer-Encoding: binary

GET Customers('ALFKI') HTTP/1.1


--batch_36522ad7-fc75-4b56-8c71-56071383e77b
Content-Type: application/http
Content-Transfer-Encoding: binary

GET Products HTTP/1.1


--batch_36522ad7-fc75-4b56-8c71-56071383e77b--"
`;
